docs/source/howto/write_workflows.rst

```diff
@@ -135,171 +133,171 @@ The second argument is the result of the work chain, extracted from the ``Int``
-    For example, the ``run_get_node`` function from the AiiDA engine returns both the result of the workflow and the work function node.
-    See the :ref:`corresponding topics section for more details <topics:processes:usage:launching>`.
+In many cases it is convenient for work chains to expose the inputs of the subprocesses it wraps so users can specify these inputs directly, as well as exposing some of the outputs produced as one of the results of the parent work chain.
+For the simple example presented in the previous section, simply copy-pasting the input and output port definitions of the subprocess ``MultiplyAddWorkChain`` was not too troublesome.
+However, this quickly becomes tedious and error-prone once you start to wrap processes with quite a few more inputs.
 
-Work chains
------------
-
-To launch a work chain, you can either use the ``run`` or ``submit`` functions.
-For either function, you need to provide the class of the work chain as the first argument, followed by the inputs as keyword arguments.
-Using the ``run`` function, or "running", a work chain means it is executed in the same system process as the interpreter in which it is launched:
+To prevent the copy-pasting of input and output specifications, the :class:`~aiida.engine.processes.process_spec.ProcessSpec` class provides the :meth:`~plumpy.ProcessSpec.expose_inputs` and :meth:`~plumpy.ProcessSpec.expose_outputs` methods.
+Calling :meth:`~plumpy.ProcessSpec.expose_inputs` for a particular ``Process`` class, will automatically copy the inputs of the class into the inputs namespace of the process specification:
 
 .. code-block:: python
 
-    from aiida.engine import run
-    from aiida.plugins import WorkflowFactory, DataFactory
-    Int = DataFactory('int')
-    MultiplyAddWorkChain = WorkflowFactory('arithmetic.multiply_add')
-
-    add_code = load_code(label='add')
-
-    results = run(MultiplyAddWorkChain, x=Int(2), y=Int(3), z=Int(5), code=add_code)
-
-Alternatively, you can first construct a dictionary of the inputs, and pass it to the ``run`` function by taking advantage of `Python's automatic keyword expansion <https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists>`_:
+    @classmethod
+    def define(cls, spec):
+        """Specify inputs and outputs."""
+        super().define(spec)
+        spec.expose_inputs(MultiplyAddWorkChain)  # Expose the inputs instead of copying their definition
+        spec.outline(
+            cls.multiply_add,
+            cls.is_even,
+        )
+        spec.output('is_even', valid_type=Bool)
+
+Be aware that any inputs that already exist in the namespace will be overridden.
+To prevent this, the method accepts the ``namespace`` argument, which will cause the inputs to be copied into that namespace instead of the top-level namespace.
```

Comment:
Is there a way to provide a link to an argument (`namespace`) the way you do to methods and classes, i.e. `` :meth:`~plumpy.ProcessSpec.expose_inputs` ``?

Reply:
There seems to be an extension for this:

https://pypi.org/project/sphinx-paramlinks/

Not sure if we want to add it though. I suppose they can just click on the method and check out the parameter. 🙃